### PR TITLE
6240/Add profile type filter = seller

### LINKED ISSUE #6240: source-amazon-ads: not working with Profile type = agency
## Enviroment
- **Source Connector and version**: source_amazon_ads (0.1.0)
- **Step where error happened**: Setup new connection

## Current Behavior

Context: Using my API tokens, a GET to https://advertising-api.amazon.com/v2/profiles returns 2 different profiles:
1. `Profile.accountInfo.type` = `seller`
2. `Profile.accountInfo.type` = `agency`

If the `Profile.accountInfo.type` is `agency`, then requests to https://advertising-api.amazon.com/v2/sp/campaigns or https://advertising-api.amazon.com/v2/portfolios/extended and also other REST endpoints are failing with the following error message:

```
{
  "code": "400",
  "details": " The request input is invalid. Advertiser ID cannot be decrypted",
  "requestId": "<omitted>"
}
```

For the `Profile.accountInfo.type` = `seller` everything works as expected.

According to the [Amazon Ads documentation](https://advertising.amazon.com/API/docs/en-us/reference/2/profiles#/Profiles/listProfiles), there's also another profile type - `vendor`. 

## Expected Behavior

`Profile.accountInfo.type` = `agency` Profiles should be skipped, or `profileTypeFilter` should be added as a configuration variable.

## Logs
*If applicable, please upload the logs from the failing operation. 
For sync jobs, you can download the full logs from the UI by going to the sync attempt page and 
clicking the download logs button at the top right of the logs display window.*

<details>
<summary>LOG</summary> 

```

{"type": "LOG", "log": {"level": "ERROR", "message": "Encountered an exception while reading stream SourceAmazonAds\nTraceback (most recent call last):\n  File \"/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/.venv/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py\", line 112, in read\n    yield from self._read_stream(\n  File \"/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/.venv/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py\", line 147, in _read_stream\n    for record in record_iterator:\n  File \"/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/.venv/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py\", line 219, in _read_full_refresh\n    for record in records:\n  File \"/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/common.py\", line 219, in read_records\n    yield from super().read_records(*args, **kvargs)\n  File \"/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/.venv/lib/python3.9/site-packages/airbyte_cdk/sources/streams/http/http.py\", line 326, in read_records\n    yield from self.parse_response(response, stream_state=stream_state, stream_slice=stream_slice)\n  File \"/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/common.py\", line 143, in parse_response\n    response.raise_for_status()\n  File \"/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/.venv/lib/python3.9/site-packages/requests/models.py\", line 953, in raise_for_status\n    raise HTTPError(http_error_msg, response=self)\nrequests.exceptions.HTTPError: 400 Client Error:  for url: https://advertising-api.amazon.com/v2/sp/campaigns?count=100\n"}}
Traceback (most recent call last):
  File "/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/main.py", line 33, in <module>
    launch(source, sys.argv[1:])
  File "/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/.venv/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 125, in launch
    for message in source_entrypoint.run(parsed_args):
  File "/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/.venv/lib/python3.9/site-packages/airbyte_cdk/entrypoint.py", line 116, in run
    for message in generator:
  File "/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/.venv/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 121, in read
    raise e
  File "/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/.venv/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 112, in read
    yield from self._read_stream(
  File "/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/.venv/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 147, in _read_stream
    for record in record_iterator:
  File "/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/.venv/lib/python3.9/site-packages/airbyte_cdk/sources/abstract_source.py", line 219, in _read_full_refresh
    for record in records:
  File "/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/common.py", line 219, in read_records
    yield from super().read_records(*args, **kvargs)
  File "/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/.venv/lib/python3.9/site-packages/airbyte_cdk/sources/streams/http/http.py", line 326, in read_records
    yield from self.parse_response(response, stream_state=stream_state, stream_slice=stream_slice)
  File "/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/common.py", line 143, in parse_response
    response.raise_for_status()
  File "/Users/dev/dev/airbyte/airbyte-integrations/connectors/source-amazon-ads/.venv/lib/python3.9/site-packages/requests/models.py", line 953, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error:  for url: https://advertising-api.amazon.com/v2/sp/campaigns?count=100

```

</details>

## Are you willing to submit a PR?

Works with the profileTypeFilter added to the request path: 
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/profiles.py#L43
```diff
- return "v2/profiles"
+ return "v2/profiles?profileTypeFilter=seller"
```

### PR DESCRIPTION
**Pull request by mid**
## What
One of the ways to solve the #6240 issue.

## How
Adding profileTypeFilter to /v2/profiles request path

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Hi @mid sorry not review this earlier. Tomorrow we're going to do it. I'm not familiar with Amazon Ads so I requested to @avida

File: airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/profiles.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ class Profiles(AmazonAdsStream):
     model = Profile
 
     def path(self, **kvargs) -> str:
-        return "v2/profiles"
+        return "v2/profiles?profileTypeFilter=seller"
```
Comment (avida) [inline airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/profiles.py:43]: please add also vendor parameter:
```python
 return "v2/profiles?profileTypeFilter=seller,vendor"
```

**Review by avida — CHANGES_REQUESTED** [review]
Havent chance to test it with agency profiles but we need vendor profile as well. Please add it to the filters parameter

File: airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/profiles.py (hunk 1 above)
Comment (mid) [reply airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/profiles.py]: done!

**Review by avida — APPROVED** [review]
Thanks, good job

**Comment by avida** [comment]
@mid  Merged, Ive created PR to bump connector version https://github.com/airbytehq/airbyte/pull/6367, image will be available when publish job completed

**Comment by mid** [comment]
> @mid Merged, Ive created PR to bump connector version #6367, image will be available when publish job completed

Thanks!

**Pull request merged**